CMakeLists.txt

```diff
@@ -1,54 +1,168 @@
 cmake_minimum_required(VERSION 3.13)
 
 get_filename_component(root ${CMAKE_CURRENT_LIST_DIR} ABSOLUTE)
-include(${root}/deps/readies/cmake/main)
+get_filename_component(binroot ${CMAKE_CURRENT_BINARY_DIR}/.. ABSOLUTE)
 
+# Platform detection
+if(APPLE)
+  set(OS "macos")
+elseif(UNIX)
+  set(OS "linux")
+elseif(WIN32)
+  set(OS "windows")
+endif()
 
-get_filename_component(binroot ${CMAKE_CURRENT_BINARY_DIR}/.. ABSOLUTE)
+# CMake policies
+if(POLICY CMP0048)
+  cmake_policy(SET CMP0048 NEW) # project() command manages VERSION variables
+endif()
+if(POLICY CMP0075)
+  cmake_policy(SET CMP0075 NEW) # Include file check macros honor CMAKE_REQUIRED_LIBRARIES
+endif()
+if(POLICY CMP0077)
+  cmake_policy(SET CMP0077 NEW) # option() honors normal variables
+endif()
 
-#----------------------------------------------------------------------------------------------
+# Initialize build flags
+set(CMAKE_LD_FLAGS "-pthread -shared -Bsymbolic -Bsymbolic-functions")
+
+# Always use .so extension even on macOS
+set(CMAKE_SHARED_LIBRARY_SUFFIX ".so")
+
+# Define compiler setup function
+function(setup_cc_options)
+  # Common compiler flags
+  set(CMAKE_C_FLAGS "-fPIC -pthread -fno-strict-aliasing -Wno-unused-function -Wno-unused-variable -Wno-sign-compare -Werror=incompatible-pointer-types -Werror=implicit-function-declaration -fcommon" PARENT_SCOPE)
+  set(CMAKE_CXX_FLAGS "-fPIC -pthread -fno-strict-aliasing -Wno-unused-function -Wno-unused-variable -Wno-sign-compare -std=c++20" PARENT_SCOPE)
```

Comment:
I think for better readability to have c++ standard as a dedicated cmake command